documentation/src/docs/table-api.md

```diff
@@ -49,43 +49,65 @@ pravega.tableSourceReaderBuilder()
 
  // (Option-1) Streaming Source
 StreamExecutionEnvironment execEnvRead = StreamExecutionEnvironment.getExecutionEnvironment();
-StreamTableEnvironment tableEnv = TableEnvironment.getTableEnvironment(execEnvRead);
+StreamTableEnvironment tableEnv = StreamTableEnvironment.create(execEnvRead);
 
-StreamTableDescriptor desc = tableEnv.connect(pravega)
-        .withFormat(new Json().failOnMissingField(true).deriveSchema())
-        .withSchema(schema)
-        .inAppendMode();
+ConnectTableDescriptor desc = tableEnv.connect(pravega)
+                .withFormat(new Json().failOnMissingField(true))
+                .withSchema(schema)
+                .inAppendMode();
 
-final Map<String, String> propertiesMap = DescriptorProperties.toJavaMap(desc);
+final Map<String, String> propertiesMap = desc.toProperties();
 final TableSource<?> source = TableFactoryService.find(StreamTableSourceFactory.class, propertiesMap)
         .createStreamTableSource(propertiesMap);
 
-tableEnv.registerTableSource("MyTableRow", source);
+String tableSourcePath = tableEnv.getCurrentDatabase() + "." + "MyTableRow";
+
+ConnectorCatalogTable<?, ?> connectorCatalogSourceTable = ConnectorCatalogTable.source(source, false);
+
+tableEnv.getCatalog(tableEnv.getCurrentCatalog()).get().createTable(
+                ObjectPath.fromString(tableSourcePath),
+                connectorCatalogSourceTable, false);
+
+// This method is deprecated but still can be used in Flink1.10 and will be removed in Flink1.11 

```

Comment:
Although this can be removed, `tableEnv.registerTableSource` seems still there in Flink 1.11, `ConnectTableDescriptor.registerTableSource` is removed in 1.11